src/System.IO.FileSystem/src/System/IO/FileSystem.Win32.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        public static void Encrypt(string path)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            string fullPath = Path.GetFullPath(path);
+
+            if (!Interop.Kernel32.EncryptFile(fullPath))
+            {
+                ThrowException(fullPath);
+            }
+        }
+
+        public static void Decrypt(string path)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            string fullPath = Path.GetFullPath(path);
+
+            if (!Interop.Kernel32.DecryptFile(fullPath))
+            {
+                ThrowException(fullPath);
+            }
+        }
+
+        private static void ThrowException(string fullPath)
+        {
+            int errorCode = Marshal.GetLastWin32Error();
+            if (errorCode == Interop.Errors.ERROR_ACCESS_DENIED)
```

Comment:
The ERROR_ACCESS_DENIED was determined empirically? I don't see it mentioned in the docs.

Reply:
I can't help with that one. The [old code](https://referencesource.microsoft.com/#mscorlib/system/io/file.cs,364) had it so presumably...

Reply:
Oh well at least the behavior will be consistent with the Fx ...